api/app/events/status.py

```diff
@@ -59,7 +59,7 @@ def get_event_status(event_id, user_id):
     else:
         application_status = 'Not Submitted'
 
-    outcome = outcome_repository.get_latest_by_user_for_event(user_id, event_id)
+    outcome = outcome_repository.get_latest_by_user_for_event_response(user_id,None, event_id)
```

Comment:
Having the response_id sometimes hard-coded to None is a dangerous practice that can lead to bugs. If we're going to link outcomes to responses, then we need to do it properly, which means:
- response_id should not be nullable
- there should always be a valid response_id when querying for an outcome.